x-pack/plugin/metering-ix/src/main/java/org/elasticsearch/xpack/meter/ix/MeterIXPoller.java

```diff
@@ -0,0 +1,153 @@
+    Scheduler.ScheduledCancellable next = null;
+    private final ReentrantLock mutex = new ReentrantLock();
+
+    public MeterIXPoller(ThreadPool threadPool, IndicesService indicesService) {
+        this.threadPool = threadPool;
+        this.indicesService = indicesService;
+    }
+
+    @Override
+    public void close() {
+        mutex.lock();
+        try {
+            if (state != Lifecycle.State.STOPPED && state != Lifecycle.State.CLOSED) {
+                state = Lifecycle.State.CLOSED;
+                if (next != null) {
+                    next.cancel();
+                }
+                next = null;
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    @Override
+    public Lifecycle.State lifecycleState() {
+        return null;
+    }
+
+    @Override
+    public void addLifecycleListener(LifecycleListener listener) {
+
+    }
+
+    @Override
+    public void start() {
+        mutex.lock();
+        try {
+            if (state == Lifecycle.State.INITIALIZED) {
+                state = Lifecycle.State.STARTED;
+                setScheduleNext();
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    @Override
+    public void stop() {
+        mutex.lock();
+        try {
+            if (state != Lifecycle.State.STOPPED && state != Lifecycle.State.CLOSED) {
+                state = Lifecycle.State.STOPPED;
+                if (next != null) {
+                    next.cancel();
+                }
+                next = null;
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    private void setScheduleNext() {
+        next = threadPool.schedule(this::logAndSchedule, TimeValue.timeValueSeconds(20), ThreadPool.Names.GENERIC);
+    }
+
+    public void logAndSchedule() {
+        logIndexStats();
+        mutex.lock();
+        try {
+            if (state == Lifecycle.State.STARTED) {
+                setScheduleNext();
+            }
+        } finally {
+            mutex.unlock();
+        }
+    }
+
+    public void logIndexStats() {
+        logger.warn("[IX] logIndexStats");
+        collectRecords();
+        logger.warn("[IX] done logIndexStats");
+    }
+
+    public List<IXRecord> collectRecords() {
+        List<IXRecord> records = new ArrayList<>();
+        for (final IndexService indexService : indicesService) {
+            String index = indexService.index().getName();
+            for (final IndexShard shard : indexService) {
+                Engine engine = shard.getEngineOrNull();
+                if (engine == null) {
+                    continue;
+                }
+                long term = shard.getOperationPrimaryTerm();
+                int shardId = shard.shardId().id();
+                for (final SegmentCommitInfo commitInfo : engine.getLastCommittedSegmentInfos()) {
+                    try {
+                        var record = new IXRecord(
```

Comment:
do we need shardId, term etc or even index to report ix size?
when checking https://github.com/elastic/mx-team/blob/main/teams/billing/services/usage_record_schema_v2.md 
there isn't expectation for all the details?
as per that email in the past, perhaps even index is not necessary
https://groups.google.com/u/1/a/elastic.co/g/es-core-infra/c/8yNT08XlEPo?pli=1